.github/workflows/codecov.yaml

```diff
@@ -18,7 +18,7 @@ jobs:
     runs-on: zephyr_runner
     needs: codecov-prep
     container:
-      image: ghcr.io/zephyrproject-rtos/ci:v0.23.3
+      image: ghcr.io/zephyrproject-rtos/ci-internal:v0.23.3.0
```

Comment:
We should not use internal/private  docker images for the common and major CI workflows. We need to a better understanding of why we are doing this and what is the impact on users trying to reproduce an issue that happend in CI without having to go install 3rd part stuff that is not part of the documented environemt or the public docker images available for everyone. Marking this to be discussed in the TSC this week.

Reply:
> what is the impact on users trying to reproduce an issue that happend in CI

The "private/internal CI docker image" is what contains the tools (in this case, the ARM FVP emulator) that cannot be included in the public CI image due to [the licensing issues](https://github.com/zephyrproject-rtos/zephyr/issues/43852#issuecomment-1068928542) (redistribution is not allowed).

The ARM FVP emulator included in this image is publicly available and free for everyone to download (only redistribution is not allowed), so anyone can go on the ARM website, download the ARM FVP, and reproduce any issues seen in the CI if ever needed; moreover, they have now changed the download process such that registration is no longer necessary.

> that is not part of the documented environemt

The instructions for downloading the ARM FVP is [documented for the boards that require the ARM FVP](https://github.com/zephyrproject-rtos/zephyr/blame/30cd0d4338601ffd39d8f5d1a992cb4f090ced6e/boards/arm/fvp_baser_aemv8r_aarch32/doc/index.rst#L17-L19).

> We need to a better understanding of why we are doing this 

There is a great value in including the ARM FVP in the CI because various ARM architecture variants and reference platforms require this tool for emulation (and, subsequently, for testing); for example, "ARMv8-R architecture port" is completely untested in the CI without the ARM FVP and we have seen various breakages in the upstream code because of this.

> Marking this to be discussed in the TSC this week.

In essence, we need to decide between:

a) having the upstream ARM architecture ports and reference platforms well tested in the CI at the cost of the minor inconvenience for a small number of users to manually download and install the ARM FVP if they ever need it for local testing

b) not testing the upstream ARM architecture ports and reference platforms in the CI and letting them rot away as we have previously seen

Given the importance of the ARM architecture support to many stakeholders and the project itself, I think the importance of this does not need to be reiterated.

Moreover, the private/internal CI docker image would eventually be something inevitable as we start adding more third-party tooling support anyway (e.g. other third-party emulators and toolchains), but let us leave that discussion for another day.

p.s. It would certainly require further discussion for the private tools that requires registration and approval like, for instance, the nSIM (for ARC); but, in the case of the ARM FVP, it is both public and freely available.